File: core/src/main/java/glitch/core/api/json/OrdinalList.java
package glitch.core.api.json;

import com.google.common.collect.ImmutableList;
import com.google.gson.annotations.SerializedName;

public interface OrdinalList<E extends IDObject<?>> {
    @SerializedName(value = "data", alternate = {
            "rooms"
    })
    ImmutableList<E> getData();

    @SerializedName("_total")
    int size();
}

File: kraken/src/main/kotlin/io/glitchlib/v5/internal/model/json/ordinal_list.kt
package io.glitchlib.v5.internal.model.json

import com.google.gson.annotations.SerializedName
import io.glitchlib.model.OrdinalList
import io.glitchlib.v5.model.json.Cheermote
import io.glitchlib.v5.model.json.User

data class Cheermotes(
        @SerializedName("actions")
        override val data: List<Cheermote>
) : OrdinalList<Cheermote>

data class Editors(
        @SerializedName("users")
        override val data: List<User>
) : OrdinalList<User>